pkg/runtime/wfengine/wfengine_test.go

```diff
@@ -368,22 +397,96 @@ func TestConcurrentActivityExecution(t *testing.T) {
 	})
 
 	ctx := context.Background()
-	client := startEngine(ctx, r)
+	client, engine := startEngine(ctx, r)
+	for _, opt := range GetTestOptions() {
+		t.Run(opt(engine), func(t *testing.T) {
+			id, err := client.ScheduleNewOrchestration(ctx, "ActivityFanOut")
+			if assert.NoError(t, err) {
+				metadata, err := client.WaitForOrchestrationCompletion(ctx, id)
+				if assert.NoError(t, err) {
+					assert.True(t, metadata.IsComplete())
+					assert.Equal(t, `["9","8","7","6","5","4","3","2","1","0"]`, metadata.SerializedOutput)
+
+					// Because all the activities run in parallel, they should complete very quickly
+					assert.Less(t, metadata.LastUpdatedAt.Sub(metadata.CreatedAt), 3*time.Second)
```

Comment:
Is this critical? I think checks like this are best left to performance tests to avoid flaky tests that are validating correctness. Thoughts?

Reply:
I agree we want to avoid tests that involve timing (since they run the risk of becoming flaky) and I debated with myself on whether to include this specific validation. In my personal deliberation, it met the "correctness" validation bar because the ability to execute activities in parallel is foundational, IMO. I would like to keep it if there's no major objection.